terraform/azure/netaddress.tf

```diff
@@ -0,0 +1,148 @@
+# This file is used to calculate and store in some locals (local variables)
+# the IP addresses of all the machines.
+
+locals {
+  ###################################################
+  ###################################################
+  # R A N G E s
+  ###################################################
+  ###################################################
+  # If vnet_name is not defined, user is supposed also to provide vnet_address_range.
+  # Both the name and the range will be then used to create the vnet.
+  # 
+  # User has the freedom to use an existing vnet. When doing so user has the freedom to both
+  # provide or not provide the associated range.
+  # If vnet_name is defined, and the vnet_address_range is empty,
+  # Terraform will try to get the ip range from the real vnet using the data source.
+  # If vnet_address_range is provided by the user, Terraform will use it
+  vnet_address_range = (
+    var.vnet_name == "" ?
+    var.vnet_address_range :
+    var.vnet_address_range == "" ? data.azurerm_virtual_network.mynet.0.address_space.0 : var.vnet_address_range
+  )
+
+  subnet_address_range = (
+    var.subnet_name == "" ?
+    (var.subnet_address_range == "" ? cidrsubnet(local.vnet_address_range, 3, 1) : var.subnet_address_range) :
```

Comment:
`3` in here it is an important change comparing with the original version that was using `8`